src/utils.ts

```diff
@@ -44,103 +42,34 @@ SankeyDiagramLink
 
 const SelectedClassName: string = "selected";
 
-export function getFillOpacity(
-    selected: boolean,
-    highlight: boolean,
-    hasSelection: boolean,
-    hasPartialHighlights: boolean): boolean {
-
-    if ((hasPartialHighlights && !highlight) || (hasSelection && !selected)) {
-        return true;
-    }
-
-    return false;
-}
-
-export function isTheDataPointNode(dataPoint: SankeyDiagramLink | SankeyDiagramNode): boolean {
-    const node: SankeyDiagramNode = <SankeyDiagramNode>dataPoint;
-
-    return node.selectableDataPoints && node.selectableDataPoints.length
-        ? true
-        : false;
-}
-
-export function isDataPointSelected(dataPoint: SankeyDiagramLink | SankeyDiagramNode): boolean {
-    const node: SankeyDiagramNode = <SankeyDiagramNode>dataPoint,
-        link: SankeyDiagramLink = <SankeyDiagramLink>dataPoint;
-
-    let selected: boolean;
-
-    if (isTheDataPointNode(dataPoint)) {
-        node.selectableDataPoints.forEach((selectableDataPoint: SelectableDataPoint) => {
-            selected = selected || selectableDataPoint.selected;
-        });
-    } else if (link.identity) {
-        selected = link.selected;
-    }
-
-    return selected;
-}
-
-export function updateFillOpacity(
-    selection: Selection<SankeyDiagramNode | SankeyDiagramLink>,
-    interactivityService?: IInteractivityService<SelectableDataPoint>,
-    hasSelection: boolean = false): void {
-
-    let hasHighlights: boolean = false;
-
-    if (interactivityService) {
-        hasHighlights = interactivityService.hasSelection();
-    }
-
-    selection.classed(SelectedClassName, (dataPoint: SankeyDiagramLink | SankeyDiagramNode): boolean => {
-        const dataPointSelected: boolean = isDataPointSelected(dataPoint),
-            theDataPointNode: boolean = isTheDataPointNode(dataPoint);
-
-        const selected: boolean = !theDataPointNode && hasSelection
-            ? !dataPointSelected
-            : dataPointSelected;
-
-        return getFillOpacity(
-            selected,
-            false,
-            hasSelection,
-            !selected && hasHighlights);
+export function isNodeSelected(node: SankeyDiagramNode, selectionManager: ISelectionManager): boolean {
+    let isSelected: boolean = false;
+    node.linkSelectableIds.forEach((selectableId: ISelectionId) => {
+        if (selectionManager.getSelectionIds().some((id: ISelectionId) => id.equals(selectableId))){
```

Comment:
const selectedIds = selectionManager.getSelectionIds(); // call it only once
 node.linkSelectableIds.forEach((selectableId: ISelectionId) => {
    if (selectedIds.includes(selectableId))){ // use _includes_ instead of _some_